Libraries/LibIPC/TransportSocketWindows.cpp

```diff
@@ -100,10 +100,25 @@ ErrorOr<void> TransportSocketWindows::duplicate_handles(Bytes bytes, Vector<size
     return {};
 }
 
-ErrorOr<void> TransportSocketWindows::transfer(Bytes bytes_to_write, Vector<size_t> const& handle_offsets)
+struct MessageHeader {
+    u32 size { 0 };
+};
+
+ErrorOr<void> TransportSocketWindows::transfer_message(Bytes bytes, Vector<size_t> const& handle_offsets)
 {
-    TRY(duplicate_handles(bytes_to_write, handle_offsets));
+    TRY(duplicate_handles(bytes, handle_offsets));
+
+    Vector<u8> message_buffer;
+    message_buffer.resize(sizeof(MessageHeader) + bytes.size());
```

Comment:
This looks kinda odd. Why not duplicate the handles into the message_buffer post-resize, rather than duplicating them into the caller's bytes and then copying the caller's bytes into a buffer? Feels like the arg could be readonly.